letterpad/letterpad
# Slate editor and plugins

Earlier we were using Quill Editor for writing stories. We have now replaced this with `Slate` editor. The UI has a big improvement in this PR. A lot of elements has been removed to make it more clean. 

We are currently supporting the below plugins:
- Bold
- Itallic
- Underline
- Image
- Headings
- CodeBlock
- Highlight
- Markdown
- Lists (Ordered and Unordered)
- LineBreak
- Quote
- Link

### Testing Procedures
The default theme `hugo` will break for Codeblocks. Earlier we used `highlight.js` and now we are using `Prism.js`. This change will be done in the theme but I will wait for the QA to approve this.

The markdown is inbuilt along with the rich text editor. So you will be able to use the rich text as well the markdown on the fly. All features of markdown should be in sync with rich text, meaning everything that works in markdown should work with rich text and vice versa.

Also, after saving the post and then reloading the browser should persist the formatting.

Thanks to @boopathi for helping me out with enhancements. The branch had a problem with merging and I had to create a new branch to resolve this, so your commits got missed out.


#### Markdown cheatsheet:
```

# H1
## H2
### H3
#### H4
##### H5
###### H6

*italic*
**bold**
_underlined_

- UnorderedList (followed by space)
* UnorderedList (followed by space)

1. OrderedList (followed by space)

To create nested list, hit tab.

Links:
[I'm an inline-style link](https://www.google.com)

Image:

![I'm an image](https://img00.deviantart.net/95e7/i/2014/007/d/3/google_abstract_by_dynamicz34-d718hzj.png)


Inline `code` has `back-ticks around` it.

> Blockquotes can be added like this.

--- (followed by a space) will add a line break.
```
>  Codeblocks has been disabled in markdown for now.
